include/sys/check.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef ZEPHYR_INCLUDE_SYS_CHECK_H_
+#define ZEPHYR_INCLUDE_SYS_CHECK_H_
+
+#include <sys/__assert.h>
+
+#if defined(CONFIG_ASSERT_ON_ERRORS)
+#define CHECK(expr) \
+	__ASSERT_NO_MSG(!(expr));   \
+	if (0)
+#elif defined(CONFIG_NO_RUNTIME_CHECKS)
+#define CHECK(...) \
```

Comment:
Are there any cases where completely dropping the arguments could generate a compiler warning about "set but not used" or something like that?